src/core/CMakeLists.txt

```diff
@@ -0,0 +1,84 @@
+find_package(LZ4)
+find_package(Boost COMPONENTS
+  system
+  program_options
+  REQUIRED)
+find_package(PkgConfig REQUIRED)
+smfc_gen(
+  CPP
+  TARGET_NAME raw_rpc_gen
+  OUTPUT_DIRECTORY ${PROJECT_SOURCE_DIR}/src/smf
+  SOURCES ${PROJECT_SOURCE_DIR}/src/core/rpc.fbs)
+
+# seastar dep
+
+if(CMAKE_BUILD_TYPE MATCHES Debug)
+  set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${SEASTAR_DIR}/build/debug")
+else()
+  set(ENV{PKG_CONFIG_PATH} "$ENV{PKG_CONFIG_PATH}:${SEASTAR_DIR}/build/release")
+endif()
+pkg_search_module(SEASTAR REQUIRED seastar)
+
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fdiagnostics-color=auto -Wall -Werror -Wextra -Wformat")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wmissing-braces -Wparentheses -Wpointer-arith")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wformat-security -Wunused -Wno-unused-parameter")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wcast-align -Wno-missing-field-initializers")
+set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wdelete-non-virtual-dtor -Wno-ignored-qualifiers")
+
+if("${CMAKE_CXX_COMPILER_ID}" STREQUAL "GNU")
+  if (CMAKE_CXX_COMPILER_VERSION VERSION_GREATER 6.1)
+    add_definitions(-DSMF_GCC_CONCEPTS=1)
+    set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -fconcepts")
+    if(CMAKE_BUILD_TYPE MATCHES Debug)
+      set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -lasan -lubsan")
+    endif()
+  endif()
+endif()
+
+
+add_library(smf STATIC
+  compression.cc
+  histogram.cc
+  histogram_seastar_utils.cc
+  random.cc
+  rpc_client.cc
+  rpc_connection_limits.cc
+  rpc_envelope.cc
+  rpc_handle_router.cc
+  rpc_letter.cc
+  rpc_recv_context.cc
+  rpc_server.cc
+  lz4_filter.cc
+  zstd_filter.cc
+  ${raw_rpc_gen})
+
+target_compile_options(smf
+  PUBLIC ${SEASTAR_CFLAGS})
+
+target_include_directories(smf
+  PUBLIC ${CMAKE_CURRENT_SOURCE_DIR}
```

Comment:
I think you'll want ${PROJECT_SOURCE_DIR}/src here to enable those linking against smf to inherit the include path that makes `#include <smf/*.h>` possible. One reason to nest `smf` in `src/include` is so that the entire `smf` tree isn't exposed through that include directive.

Reply:
that is a really good thought I hadn't considered!

Reply:
https://github.com/senior7515/smf/pull/221